spec/forms/project_form_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'rails_helper'
+
+describe ProjectForm do
+  describe "validations" do
+    context "slack channel already exists" do
+      it "adds an error to the form object" do
+        user = FactoryGirl.create(:user)
+        slack = double(find_channel_by_name: true)
+        form = ProjectForm.new(name: "name", description: "description", user: user, slack: slack, github: double)
+        form.valid?
+        expect(form.errors[:channel]).to include "name already exists.  Pick new name for your project"
+      end
+
+    end
+  end
+
+  describe "#save" do
+    context "when valid" do
+      it "calls persist!" do
+        user = FactoryGirl.create(:user)
+        form = ProjectForm.new(name: "name", description: "description", user: user, slack: slack_double, github: double)
+        expect(form).to receive(:persist!).and_return(true)
+        form.save
+      end
+    end
+
+    context "when invalid" do
+      it "returns false" do
+        form = ProjectForm.new(name: "", description: "description", slack: slack_double)
+        expect(form.save).to eq false
+      end
+    end
+  end
+
+  describe "#persist!" do
+    context "without slack/github" do
+      let(:user) { FactoryGirl.create(:user) }
+      let(:form) { ProjectForm.new(name: "sample project", description: '', user: user, slack: slack_double, github: double) }
```

Comment:
Line is too long. [126/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.